### PR TITLE
Enables git hooks automatically.

### LINKED ISSUE #4221: Enable hooks/pre-commit in git subsystem using DS script
1. Link the script to .git folder:
   `ln -s hooks/pre-commit .git/hooks/pre-commit`
2. Process errors: when file exists, directory doesn't exist, etc.

### PR DESCRIPTION
**Pull request by sergiitk**
Plus, cleans up apc.php due to deprecation.

Closes #4221.

### CODE REVIEW

File: bin/ds [hunk 1]
```diff
@@ -162,9 +180,6 @@ function build {
     source ~/.bashrc
   fi
 
-  echo 'Adding APC script at /apc.php...'
-  ln -s /srv/salt/php5/apc.php $WEB_PATH/apc.php
-
```
Comment (angaither) [inline bin/ds:167]: just making sure this was an intentional removal

Comment (sergiitk) [inline bin/ds:167]: Yep, see PR description:

> Plus, cleans up apc.php due to deprecation.

Comment (sergiitk) [inline bin/ds:167]: I don't provide this file in new Vagrant boxes. We don't use APC cache anymore, it has been replaced by Zend OPcache.

Comment (angaither) [inline bin/ds:167]: cool, I'm trying to test it now, just a little confused by all the if statements

File: bin/ds [hunk 2]
```diff
@@ -142,6 +142,24 @@ function build {
...
 
+  echo "Enabling git hooks"
+  if [[ -d "$BASE_PATH/.git/hooks/" ]]; then
+    # Pre-commit.
+
+    HOOK_PRE_COMMIT=$BASE_PATH/.git/hooks/pre-commit
+    if [[ ! -f $HOOK_PRE_COMMIT ]]; then
+      if [[ -L $HOOK_PRE_COMMIT ]]; then
+        # Symlink is broken.
+        rm $HOOK_PRE_COMMIT
```
Comment (angaither) [inline bin/ds:160]: why if it's a not a file and symlink why is it removing it?
can you add documentation to say what's going on here?

**Comment by angaither** [comment]
:+1:

**Pull request merged**